experiments/PlaywrightSharp/PlaywrightSharp/ApiSurface/IBrowserContext.cs

```diff
@@ -0,0 +1,176 @@
+	/// Playwright allows creation of "incognito" browser contexts with `browser.newContext()` method. "Incognito" browser contexts
+	/// don't write any browsing data to disk.
+	/// </summary>
+	public interface IBrowserContext
+	{
+		event EventHandler<IBrowserContext> OnClose;
+		event EventHandler<IPage> OnPage;
+		/// <summary>
+		/// Adds cookies into this browser context. All pages within this context will have these cookies installed. Cookies can be obtained
+		/// via <see cref="IBrowserContext.CookiesAsync"/>.
+		/// </summary>
+		Task AddCookiesAsync(BrowserContextCookies[] cookies);
+		/// <summary>
+		/// Adds a script which would be evaluated in one of the following scenarios:
+		/// <list>
+		/// <item><description>Whenever a page is created in the browser context or is navigated.</description></item>
+		/// <item><description>Whenever a child frame is attached or navigated in any page in the browser context. In this case, the script is evaluated in the context of the newly attached frame.</description>
+		/// </item>
+		/// </list>
+		/// The script is evaluated after the document was created but before any of its scripts were run. This is useful to amend the
+		/// JavaScript environment, e.g. to seed `Math.random`.
+		/// An example of overriding `Math.random` before the page loads:
+		/// </summary>
+		Task AddInitScriptAsync();
+		/// <summary>
+		/// Returns the browser instance of the context. If it was launched as a persistent context null gets returned.
+		/// </summary>
+		IBrowser GetBrowser();
+		/// <summary>
+		/// Clears context cookies.
+		/// </summary>
+		Task ClearCookiesAsync();
+		/// <summary>
+		/// Clears all permission overrides for the browser context.
+		/// </summary>
+		Task ClearPermissionsAsync();
+		/// <summary>
+		/// Closes the browser context. All the pages that belong to the browser context will be closed.
+		/// </summary>
+		Task CloseAsync();
+		/// <summary>
+		/// If no URLs are specified, this method returns all cookies. If URLs are specified, only cookies that affect those URLs are
+		/// returned.
+		/// </summary>
+		Task<BrowserContextCookiesResult[]> CookiesAsync(string[] urls);
+		/// <summary>
+		/// The method adds a function called {PARAM} on the `window` object of every frame in every page in the context. When called,
+		/// the function executes {PARAM} and returns a [Promise] which resolves to the return value of {PARAM}. If the {PARAM} returns a [Promise], it will be awaited.
+		/// The first argument of the {PARAM} function contains information about the caller: `{ browserContext: BrowserContext, page:
+		/// Page, frame: Frame }`.
+		/// See <see cref="IPage.ExposeBindingAsync"/> for page-only version.
+		/// An example of exposing page URL to all frames in all pages in the context:
+		/// An example of passing an element handle:
+		/// </summary>
+		Task ExposeBindingAsync(string name, Action callback, bool handle);
```

Comment:
We will need a few overloads for this callback

Reply:
We might, yes. I propose we add those overloads *when* we need them - i.e. if users _actually_ ~need~ use them :-).